server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregator.java

```diff
@@ -202,7 +202,8 @@ private Sort buildIndexSortPrefix(LeafReaderContext context) throws IOException
             return null;
         }
         List<SortField> sortFields = new ArrayList<>();
-        for (int i = 0; i < indexSort.getSort().length; i++) {
+        int max = Math.min(indexSort.getSort().length, sourceConfigs.length);
```

Comment:
You can still hit the bug if the source configs is smaller than the index sort ? Since we're looking for the biggest prefix of the index sort maybe we could leave the for loop as it is but add a check that breaks the loop if the source configs length is reached:
`````
if (i >= sourceConfigs.length) {
  break;
}
`````

Reply:
I think the way I have it covers both cases, right?

Reply:
oh right, I read the variable named `max` and thought that you were extracting the max of the length :(